packages/suite/src/components/suite/labeling/MetadataLabeling/AutoScalingInput.tsx

```diff
@@ -0,0 +1,91 @@
+import { useOnce } from '@trezor/react-utils';
+import React, { useRef, useEffect, forwardRef, useState } from 'react';
+import styled from 'styled-components';
+
+const HiddenInputToMeasurePlaceholderScrollableWidth = styled.input`
+    visibility: hidden !important;
+    width: 0 !important;
+`;
+
+type ApplyWidthParams = {
+    ref: React.MutableRefObject<HTMLInputElement | null>;
+    width: number;
+    minWidth: number;
+};
+
+const applyWidth = ({ width, minWidth, ref }: ApplyWidthParams) => {
+    if (ref?.current?.style) {
+        // First it needs to be set to minimum, browser then rescales it with content so we can use correct scrollWidth
+        ref.current.style.width = `${minWidth}px`; // See: https://stackoverflow.com/a/75227086
+        ref.current.style.width = `${width}px`;
+    }
+};
+
+type Props = React.DetailedHTMLProps<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    HTMLInputElement
+> & { minWidth: number };
+
+export const AutoScalingInput = forwardRef<HTMLInputElement, Props>(
+    ({ value, minWidth, ...props }, ref) => {
+        const [placeholderWidth, setPlaceholderWidth] = useState(0);
+        const inputRef = useRef<HTMLInputElement | null>(null);
+        const placeholderMeasureRef = useRef<HTMLInputElement | null>(null);
+
+        const calculatedMin = Math.max(minWidth, placeholderWidth);
+
+        // Measure the size of the placeholder
+        useEffect(() => {
+            if (placeholderMeasureRef?.current?.style) {
+                setPlaceholderWidth(placeholderMeasureRef.current.scrollWidth ?? 0);
+            }
+        }, [placeholderMeasureRef, setPlaceholderWidth]);
+
+        useEffect(() => {
+            const isValueEmpty = value === '' || value === null || value === undefined;
+
+            applyWidth({
+                width: calculatedMin,
+                minWidth: isValueEmpty
+                    ? calculatedMin
+                    : Math.max(calculatedMin, inputRef?.current?.scrollWidth ?? calculatedMin),
+                ref: inputRef,
+            });
+        }, [value, inputRef, calculatedMin]);
+
+        return (
+            <>
+                <HiddenInputToMeasurePlaceholderScrollableWidth
+                    className={props.className}
+                    type="text"
+                    ref={placeholderMeasureRef}
+                    value={props.placeholder}
+                    readOnly
```

Comment:
:heavy_check_mark:

Reply:
> Warning: You provided a `value` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultValue`. Otherwise, set either `onChange` or `readOnly`.

This error is shown